src/views/Layout.jsx

```diff
@@ -1,8 +1,22 @@
 import { Outlet, NavLink } from 'react-router-dom';
 import FoodPuns from '../components/FoodPuns';
+import { useState, useEffect } from 'react';
 
 import './Layout.css';
 
+function NavBarToggle() {
+	const [activeComponent, setActiveComponent] = useState();
+	useEffect(() => {
+		setActiveComponent(`${window.location.pathname.replace('/', '')}`);
+	}, [window.location.pathname]);
```

Comment:
I think this looks good, you could also do this instead of having a nested NavBarToggle component, was there a specific reason why you separated them?

```suggestion
export function Layout() {
	const [active, setActive] = useState();
	useEffect(() => {
		setActiveComponent(window.location.pathname.slice(1))}, [window.location.pathname]
		
		......
		
		return  (
		
		<div className="Layout">
				<NavLink to="/summary" className="Nav-link">
					Summary
				</NavLink>
				<div className={`Nav-link__shadow 
				Nav-link__shadow_${active}`}></div
			</nav>
		</div>
	);
		
		
		
```

Reply:
So I was thinking if the <div> component is separate that contains the state, then this component will re-render on its own, instead of triggering the entire <layout> to re-render again. I believe the <layout> component re-renders already whenever the path changes, so I wanted to avoid having it re-render again just for the <div> component to slide over... Or is that not true? @xynree

Reply:
True! In that case you might want to take out the `NavbarToggle` component entirely and keep it in a different file?